core/src/test/java/com/ibm/wala/core/tests/cha/AnalysisScopeTest.java

```diff
@@ -55,4 +62,75 @@ public void testBaseScope() throws IOException, ClassHierarchyException {
                 ClassLoaderReference.Application, "Ljava/awt/AlphaComposite")),
         "found unexpected class");
   }
+
+  @Test
+    public void testToJson() throws IOException {
+        AnalysisScope scope =
+            AnalysisScopeReader.instance.readJavaScope(
+                TestConstants.WALA_TESTDATA,
+                // new FileProvider().getFile("J2SEClassHierarchyExclusions.txt"),
+                new FileProvider().getFile("GUIExclusions.txt"),
+                AnalysisScopeTest.class.getClassLoader());
+        Gson gson = new Gson();
+        Type type = new TypeToken<LinkedHashMap<String, Object>>(){}.getType();
+        LinkedHashMap<String, Object> map = gson.fromJson(scope.toJson(), type);
+        System.out.println(map);
+        if(map.containsKey("Exclusions")) {
+            String[] exclusions = scope.getExclusions().toString().split("\\|");
+            ArrayList<String> arr2 = new ArrayList<>();
+            for(int i = 0; i < exclusions.length; i++){
+                String word = exclusions[i];
+                word = word.replace("(", "");
+                word = word.replace(")", "");
+                arr2.add(word);
+            }
+            assertEquals(arr2, map.get("Exclusions"));
+        }
+        Type type2 = new TypeToken<LinkedHashMap<String, ArrayList<String>>>(){}.getType();
+        LinkedHashMap<String, ArrayList<String>> loaders = gson.fromJson(gson.toJson(map.get("Loaders")), type2);
```

Comment:
Here I am confused, why do you need to serialize the data in the map back to JSON and then parse it again?  I think you can get rid of this and just use some downcasts to get the data you want out of the map directly